corehq/apps/users/views/__init__.py

```diff
@@ -375,6 +375,11 @@ def post(self, request, *args, **kwargs):
         else:
             return self.get(request, *args, **kwargs)
 
+    def dispatch(self, *args, **kwargs):
+        if not user_can_access_other_user(self.domain, self.request.couch_user, self.editable_user):
```

Comment:
nit: you can also do this inside the `editable_user` method by raising `PermissionDenied` if the user isn't accessible.